client/src/lib/hooks/__tests__/useConnection.test.tsx

```diff
@@ -480,6 +485,57 @@ describe("useConnection", () => {
     });
   });
 
+  describe("Ref Resolution", () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    test("resolves $ref references in requestedSchema properties before validation", async () => {
+      const mockProtocolOnMessage = jest.fn();
+
+      mockSSETransport.onmessage = mockProtocolOnMessage;
+
+      const { result } = renderHook(() => useConnection(defaultProps));
+
+      await act(async () => {
+        await result.current.connect();
+      });
+
+      const mockRequestWithRef: JSONRPCMessage = {
+        jsonrpc: "2.0",
+        id: 1,
+        method: "elicitation/create",
+        params: {
+          message: "Please provide your information",
+          requestedSchema: {
+            type: "object",
+            properties: {
+              name: {
+                $ref: "#/properties/nameDef",
+              },
+              nameDef: {
+                type: "string",
+                title: "Name",
+              },
```

Comment:
This example is a little confusing. I know it's just a test, but in real life this wouldn't happen. You wouldn't have two fields to collect called `name` and `nameDef`. When you have two properties with the same schema, one can refer to the other. I suggest refactoring to something more likely.

```suggestion
              "source": {
                "type": "string",
                "minLength": 1,
                "title": "A Connectable Node"
              },
              "target": {
                "$ref": "#/properties/source"
              }
```